semgrep-core/bin/Main.ml

```diff
@@ -405,7 +405,13 @@ let get_final_files xs =
     | Some lang -> Lang.files_of_dirs_or_files lang xs
   in
   let files = filter_files files in
-  files
+
+  let explicit_files = xs |> List.filter(fun file ->
+      Sys.file_exists file && not (Sys.is_directory file)
+    )
+  in
+
+  files @ explicit_files
```

Comment:
How does this avoid double-passing files in each side?

Reply:
FWIW, I'd think you'd make a set out of `files`, and check lack of set membership.

Reply:
Good call this doesn't dedup